pkg/controllers/machinesync/machine_sync_controller_test.go

```diff
@@ -99,23 +108,73 @@ var _ = Describe("With a running MachineSync Reconciler", func() {
 			WithGenerateName("openshift-cluster-api-").Build()
 		Expect(k8sClient.Create(ctx, capiNamespace)).To(Succeed(), "capi namespace should be able to be created")
 
-		mapiMachineBuilder = machinev1resourcebuilder.Machine().
-			WithNamespace(mapiNamespace.GetName()).
-			WithGenerateName("foo").
-			WithProviderSpecBuilder(machinev1resourcebuilder.AWSProviderSpec().WithLoadBalancers(nil))
-
 		infrastructureName := "cluster-foo"
 		capaClusterBuilder = capav1builder.AWSCluster().
 			WithNamespace(capiNamespace.GetName()).
 			WithName(infrastructureName)
 		Expect(k8sClient.Create(ctx, capaClusterBuilder.Build())).To(Succeed(), "capa cluster should be able to be created")
 
+		// Create the CAPI Cluster to have valid owner reference to it
+		capiClusterBuilder := capiv1resourcebuilder.Cluster().WithNamespace(capiNamespace.GetName()).WithName(infrastructureName)
+		Expect(k8sClient.Create(ctx, capiClusterBuilder.Build())).To(Succeed(), "capi cluster should be able to be created")
+
 		// We need to build and create the CAPA Machine in order to
 		// reference it on the CAPI Machine
 		capaMachineBuilder = capav1builder.AWSMachine().
 			WithNamespace(capiNamespace.GetName()).
 			WithName("machine-template")
 
+		mapiMachineSetBuilder = machinev1resourcebuilder.MachineSet().
+			WithNamespace(mapiNamespace.GetName()).
+			WithName("foo-machineset").
+			WithProviderSpecBuilder(machinev1resourcebuilder.AWSProviderSpec().WithLoadBalancers(nil))
+
+		// Create the MAPI MachineSet to have valid owner reference to it
+		mapiMachineSet = mapiMachineSetBuilder.Build()
+		Expect(k8sClient.Create(ctx, mapiMachineSet)).Should(Succeed())
+		ownerReferencesToMapiMachineSet = []metav1.OwnerReference{{
+			APIVersion:         machinev1beta1.GroupVersion.String(),
+			Kind:               machineSetKind,
+			Name:               mapiMachineSet.Name,
+			UID:                mapiMachineSet.UID,
+			Controller:         ptr.To(true),
+			BlockOwnerDeletion: ptr.To(true),
+		},
+		}
```

Comment:
Maybe this would be neater?
```suggestion
		ownerReferencesToMapiMachineSet = []metav1.OwnerReference{
			{
				APIVersion:         machinev1beta1.GroupVersion.String(),
				Kind:               machineSetKind,
				Name:               mapiMachineSet.Name,
				UID:                mapiMachineSet.UID,
				Controller:         ptr.To(true),
				BlockOwnerDeletion: ptr.To(true),
			},
		}
```